# Type: fix
# Subtype: I3
# English Summary: Even without the Groovy facet, Grails should still be compiled
# Chinese Summary: 即使没有Groovy facet，也要编译Grails
# Entry Number: 24

# === Git Diff Content ===

diff --git a/plugins/groovy/src/org/jetbrains/plugins/groovy/compiler/GroovyCompiler.java b/plugins/groovy/src/org/jetbrains/plugins/groovy/compiler/GroovyCompiler.java
index 2fefa32..35e8e5e 100644
--- a/plugins/groovy/src/org/jetbrains/plugins/groovy/compiler/GroovyCompiler.java
+++ b/plugins/groovy/src/org/jetbrains/plugins/groovy/compiler/GroovyCompiler.java
@@ -94,7 +94,7 @@ public class GroovyCompiler implements TranslatingCompiler {
       final Module module = entry.getKey();
       final GroovyFacet facet = GroovyFacet.getInstance(module);
       final Set<VirtualFile> moduleFiles = entry.getValue();
-      if (facet != null && facet.getConfiguration().isCompileGroovyFiles()) {
+      if (facet == null || facet.getConfiguration().isCompileGroovyFiles()) {
         doCompile(compileContext, successfullyCompiled, toRecompile, module, moduleFiles);
       } else {
         final ResourceCompiler resourceCompiler = new ResourceCompiler(myProject, CompilerConfiguration.getInstance(myProject));
